File: sdkwork-im-sdk-flutter/composed/lib/src/inbox_module.dart
import 'package:im_sdk_generated/im_sdk_generated.dart';

import 'context.dart';

class ImInboxModule {
  final ImSdkContext context;

  ImInboxModule(this.context);

  Future<InboxResponse?> list() {
    return context.transportClient.inbox.getInbox();
  }
}

File: sdkwork-im-sdk-rust/composed/src/inbox_module.rs
use crate::ImSdkContext;
use sdkwork_im_sdk_generated::{InboxResponse, SdkworkError};

#[derive(Clone)]
pub struct ImInboxModule {
  context: ImSdkContext,
}

impl ImInboxModule {
  pub(crate) fn new(context: ImSdkContext) -> Self {
    Self { context }
  }

  pub async fn get(&self) -> Result<InboxResponse, SdkworkError> {
    self.context.transport_client().inbox().get_inbox().await
  }
}

